### PR TITLE
Cam port

### PR DESCRIPTION
**Pull request by nusbaume**
Added modifications to allow for CESM and regression script builds in git.  I should note that I had to revert one of your changes to "configure" in order to get the regression scripts to work.  Also some of the white-spacing in the modified scripts can turn out a little wonky, but the code itself should be fine. 

Python script tests run for latest commit:

./svn_select_to_git.py --author-table CAM_name_table.txt --tags https://svn-ccsm-models.cgd.ucar.edu/cam1/trunk_tags --rev 91300:91422 cam_trunk_svn ../cam_trunk_git https://svn-ccsm-models.cgd.ucar.edu/cam1/trunk

test/cam_test.sh

CAM tests run for latest commit:

"test/system/test_driver.sh -f" on izumi, with cam6_1_027 tag, with NAG compiler, in CAM-only framework.

"test/system/test_driver.sh -f" on hobart, with cam_cesm2_1_rel_26 tag, with NAG compiler, in CESM framework.

Also ran "test_drivers.sh -f" with ".git" removed, ".gitignore" removed, an extra ".git" file added, and in a subversion repository, to ensure that the "TGIT.sh" script behaves as expected.

### CODE REVIEW

File: svn_select_to_git.py [hunk 1]
```diff
@@ -297,6 +297,242 @@ def cam_svn_to_git_mods(repo_dir):
...
+    patterns = {'root_dir=\"(\$\( dirname ){4}\$\{tdir\} (\)[ \"]){4}' :
+               ('trial_dir="$( dirname $( dirname $( dirname $( dirname ${tdir} ) ) ) )"\n'
+                '    if [ -d "${trial_dir}/cime/scripts" ]; then\n'
+                '      root_dir=$trial_dir\n'
+                '    else\n'
+                '      root_dir="$( dirname $( dirname ${tdir} ) )"\n'
+                '    fi\n'),
+
+                'export CAM_ROOT=\\\`cd \\\\\$\{CAM_SCRIPTDIR}(/\.\.){4} ; pwd \\\`' :
+               ('if [ -d \`${CAM_SCRIPTDIR}/../../components\` ]; then\n'
```
Comment (gold2718) [inline svn_select_to_git.py]: I believe the backtick syntax here is illegal and will generate both an error message and the wrong test result. Always use double quotes around strings in bash test expressions.

File: svn_select_to_git.py [hunk 2]
```diff
@@ -297,6 +297,242 @@ def cam_svn_to_git_mods(repo_dir):
...
+                '        export CAM_ROOT=\`cd \${CAM_SCRIPTDIR}/../.. ; pwd \`\n'
+                '    else\n'
+                '        export CAM_ROOT=\`cd \${CAM_SCRIPTDIR}/../../../.. ; pwd \`\n'
+                '    fi'),
+
+                'echo \"ERROR: unable to determine script directory \"' :
+               ('if [ -n "\${CAM_ROOT}" ] && [ -f \${CAM_ROOT}/test/system/test_driver.sh ]; then\n'
```
Comment (gold2718) [inline svn_select_to_git.py]: For consistency, please add double quotes around the second string. Also, a cleaner bash syntax is:
```
if [ -n "\${CAM_ROOT}"  -a  -f "\${CAM_ROOT}/test/system/test_driver.sh" ]; then
```

**Review by gold2718 — CHANGES_REQUESTED** [review]
Couple of minor change but I wonder if test_driver.sh is working for the wrong reasons.

**Review by gold2718 — CHANGES_REQUESTED** [review]
Please remove the TFM test. I suggest doing this by adding a pattern for tests_pretag_hobart_nag and tests_pretag_izumi_nag to remove the `fm001` line.

**Comment by gold2718** [comment]
Please add a description of the tests run to the PR description.  This should include running test/cam_test.sh and a list of tests run for both a standalone clone and a CESM-style clone.

**Comment by nusbaume** [comment]
Python script tests run for latest commit:

./svn_select_to_git.py --author-table CAM_name_table.txt --tags https://svn-ccsm-models.cgd.ucar.edu/cam1/trunk_tags --rev 91300:91422 cam_trunk_svn ../cam_trunk_git https://svn-ccsm-models.cgd.ucar.edu/cam1/trunk

test/cam_test.sh

CAM tests run for latest commit:

"test/system/test_driver.sh -f" on izumi, with cam6_1_027 tag, with NAG compiler, in CAM-only framework.

"test/system/test_driver.sh -f" on hobart, with cam_cesm2_1_rel_26 tag, with NAG compiler, in CESM framework.

Also ran "test_drivers.sh -f" with ".git" removed, ".gitignore" removed, an extra ".git" file added, and in a subversion repository, to ensure that the "TGIT.sh" script behaves as expected.

**Pull request merged**